### PR TITLE
Chore/#tc 164 add error logging to webstore

### PR DESCRIPTION
**Pull request by Kwandes**
### This pull request closes:

✔️ Closes Jira Issue: [TC-163](https://treecreate.atlassian.net/browse/TC-163)

### Which service(s) does this issue affect:

- [x] 🛰️ API
- [x] 🛒 webstore
- [x] 🎓 admin-page

### Types of changes

- [ ] 🐛 Bug fix (a non-breaking change which fixes an issue)
- [x] 🧱 New feature (a non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🤖 CI/CD (a change to the CI/CD pipeline)
- [ ] ⚗️ Refactoring (an update to some already existing code)
- [ ] 💄 Style (Markup, formatting, CSS)

### Documentation Updates

- [x] 📂 No need for updates
- [ ] 📁 Requires an update ( _not done within this issue_ )
- [ ] 🗃️ Has been updated ( _done within this issue_ )

### Testing checklist

- [x] 💪 Manual tests
- [ ] 🔧 Automatic tests

#### Browser compatibility - _**[**Frontend only**]**_

- [x] Tested on Safari
- [x] Tested on Chrome
- [ ] Tested on Mozilla Firefox

### What changes have been done within this issue?

- New model `Errorlog`
- New endpoints for fetching and creating error logs
- Webstore now sends logs for every failed HTTP request and many of the errors that occur
- New page in Admin page `/error-logs`

### CODE REVIEW

File: apps/webstore/src/app/pages/auth/collection/collection.component.ts [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import { HttpErrorResponse } from '@angular/common/http';
 import { Component, OnInit } from '@angular/core';
-import { DesignTypeEnum, IDesign } from '@interfaces';
+import { ErrorlogPriorityEnum, DesignTypeEnum, IDesign } from '@interfaces';
```
Comment (Teodor25) [inline apps/webstore/src/app/pages/auth/collection/collection.component.ts]:
```suggestion
import { ErrorlogPriorityEnum, IDesign } from '@interfaces';
```

Comment (Teodor25) [reply apps/webstore/src/app/pages/auth/collection/collection.component.ts]: There seems to be a warning that the designTypeEnum is not in use, so i guess it should be removed.

**Review by Teodor25 — APPROVED** [review]
Looks good fam

File: apps/webstore/src/app/pages/auth/collection/collection.component.ts (hunk 1 above)
Comment (Kwandes) [reply apps/webstore/src/app/pages/auth/collection/collection.component.ts]: Resolved in [9d5ad6a](https://github.com/treecreate/webstore/pull/347/commits/9d5ad6af64b711ed5dddb6f6f032eebbf2ba2b39)

**Pull request merged**